app/views/handlers/block.py

```diff
@@ -131,10 +131,13 @@ def _get_routing_path(self):
             list_item_id=self._current_location.list_item_id,
         )
 
-    def _update_section_completeness(self, location: Optional[Location] = None):
+    def _update_section_completeness(
+        self, location: Optional[Location] = None, routing_path: Optional = None
+    ):
```

Comment:
This method shouldn't be modified to take any routing path. Everything in this class works at a block level and assumes it's the current location (unless it's a parent location). 

We should consider moving the `section_status` calculation to `questionnaire_store_updater.update_section_status`, so this method would look more like:

```python
    def _update_section_completeness(
        self, location: Optional[Location] = None
    ):
        location = location or self._current_location

        self.questionnaire_store_updater.update_section_status(
            is_complete=self.path_finder.is_path_complete(self._routing_path),
            section_id=location.section_id,
            list_item_id=location.list_item_id,
        )
```